src/boot/primary.js

```diff
@@ -0,0 +1,220 @@
+        } else {
+            try {
+                itemToSendBack = JSON.parse(retrievedItem);
+            } catch (error) {
+                console.info('Error retrieving', item, 'from storage. Parsing error:', error);
+            }
+        }
+
+        return itemToSendBack;
+    }
+
+    app.mixin({
+        methods: {
+            checkNeedsTokenRefresh: checkNeedsTokenRefresh,
+            attemptRefreshToken: attemptRefreshToken,
+            parseFromStorage: parseFromStorage
+        }
+    });
+
+    // STORE FUNCTIONS
+
+    function initStore () {
+        const metaverseConfigItems = {
+            // Metaverse Config
+            server: localStorage.getItem('metaverseConfig.server') // string
+        };
+
+        console.info('Initing metaverseConfig.server', metaverseConfigItems.server);
+
+        if (metaverseConfigItems.server !== null) {
+            console.info('metaverseConfigItems.server !== null');
+            store.commit('mutate', {
+                update: true,
+                property: 'metaverseConfig',
+                with: {
+                    server: metaverseConfigItems.server
+                }
+            });
+        }
+
+        const accountItems = {
+            // Account / General Stuff
+            isLoggedIn: parseFromStorage('isLoggedIn'), // bool
+            isAdmin: parseFromStorage('isAdmin'), // bool
+            useAsAdmin: parseFromStorage('useAsAdmin'), // bool
+            username: localStorage.getItem('username'), // string
+            accountId: localStorage.getItem('accountId'), // string
+            metaverseServer: localStorage.getItem('metaverseServer'), // string
+            // Account / Token Stuff
+            accessToken: localStorage.getItem('accessToken'), // string
+            refreshToken: localStorage.getItem('refreshToken'), // string
+            tokenType: localStorage.getItem('tokenType'), // string
+            createdAt: parseFromStorage('createdAt'), // int
+            expiresIn: parseFromStorage('expiresIn'), // int
+            scope: localStorage.getItem('scope') // string
+        };
+
+        store.commit('mutate', {
+            property: 'account',
+            with: accountItems
+        });
+
+        // const placesItems = {
+        //     // Places / General Stuff
+        //     showOnlyMine: parseFromStorage('showOnlyMine') // bool
+        // };
+        //
+        // store.commit('mutate', {
+        //     property: 'places',
+        //     with: placesItems
+        // });
+
+        const dashboardConfig = {
+            dashboardTheme: parseFromStorage('dashboardTheme') // int
+        };
+
+        store.commit('mutate', {
+            property: 'dashboardConfig',
+            with: dashboardConfig
+        });
+
+        store.commit('mutate', {
+            property: 'initialized',
+            with: true
+        });
+    }
+
+    // END STORE FUNCTIONS
+
+    // ROUTER CONTROLS
+
+    router.beforeEach((to, from, next) => {
```

Comment:
Are we even using router at this point?

Reply:
Uhh, at the moment, no, but we may later. I want to migrate all the "metaverse dashboard" components to this app, so at some point it may come in handy. Though, that all depends on how well it plays with our very custom and dynamic ways of doing things.